InternetStoreCHIReloaded/DAL/Repositories/UsersRepository.cs

```diff
@@ -0,0 +1,220 @@
+
+                response.Message = "Errors occured while creating user:";
+                foreach (var msg in userCreateResult.Errors)
+                {
+                    response.Message += $"\n{msg.Description}";
+                }
+
+                return response;
+            }
+        }
+
+        public async Task<DbResponse> SetProductToUserCartAsync(int userId, ProductToCartDbModel productToCartDbModel)
+        {
+            var user = await GetUserWithCartAsync(userId);
+
+            if (user.UserCart.CartItems.Any(ci => ci.ProductId == productToCartDbModel.ProductId))
+            {
+                var product = user.UserCart.CartItems.Where(ci => ci.ProductId == productToCartDbModel.ProductId).Single();
+                product.Quantity = productToCartDbModel.Quantity;
+            }
+            else
+            {
+                user.UserCart.CartItems.Add(new ProductWithQuantityEntity()
+                {
+                    ProductId = productToCartDbModel.ProductId,
+                    Quantity = productToCartDbModel.Quantity,
+                });
+            }
+
+            bool saveSuccess = await SaveAsync();
+
+            var dbResponse = new DbResponse();
+
+            if (saveSuccess)
+            {
+                dbResponse.IsSuccessful = true;
+                dbResponse.Message = "Product was set successfully!";
+            }
+            else
+            {
+                dbResponse.IsSuccessful = false;
+                dbResponse.Message = "Nothing has changed.";
+            }
+
+            return dbResponse;
+        }
+
+        public async Task<DbResponse> RemoveProductSetFromUserCartAsync(int userId, int productId)// removes entry from ProductWithQuantity table
+        {
+            var user = await GetUserWithCartAsync(userId);
+
+            DbResponse dbResponse = new DbResponse();
+
+            var productInCart = user.UserCart.CartItems.Where(p => p.ProductId == productId).FirstOrDefault();
+            if (productInCart != null)
+            {
+                user.UserCart.CartItems.Remove(productInCart);
+                _dbcontext.ProductsWithQuantity.Remove(productInCart);// also remove pwq entry
+
+                bool success = await UpdateAsync(user);
+
+                if (success)
+                {
+                    dbResponse.IsSuccessful = true;
+                    dbResponse.Message = "Product entry was removed successfuly!";
+                }
+                else
+                {
+                    dbResponse.IsSuccessful = false;
+                    dbResponse.Message = "Nothing has changed.";
+                }
+            }
+            else
+            {
+                dbResponse.IsSuccessful = false;
+                dbResponse.Message = "Product was not found in cart";
+            }
+
+            return dbResponse;
+        }
+
+        public async Task<DbResponse<int>> GetQuantityOfProductInCartAsync(int userId, int productId)
```

Comment:
Looks like we don't need those response messages at all. This method could return simple bool result. True - item exists, false - item is absent